pkg/controller/migmigration/backup.go

```diff
@@ -281,6 +282,52 @@ func (t *Task) updateBackup(backup *velero.Backup) error {
 	return nil
 }
 
+func (t *Task) deleteBackups() error {
+	client, err := t.getSourceClient()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	labels := t.Owner.GetCorrelationLabels()
+	labels["migmigration"] = string(t.Owner.GetUID())
+	list := velero.BackupList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(labels),
+		&list)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	for _, backup := range list.Items {
+		request := buildDeleteBackupRequest(backup.Name, backup.UID)
+		if err := client.Create(context.TODO(), request); err != nil {
+			log.Trace(err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func buildDeleteBackupRequest(name string, uid types.UID) *velero.DeleteBackupRequest {
+	return &velero.DeleteBackupRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    migapi.VeleroNamespace,
+			GenerateName: name + "-",
+			Labels: map[string]string{
+				velero.BackupNameLabel: label.GetValidName(name),
+				velero.BackupUIDLabel:  string(uid),
```

Comment:
What are these labels for?

Reply:
These are recognized by velero, and it is a copy of a method from  https://github.com/vmware-tanzu/velero/blob/3e1b8e01a7d96f84ff9016445d5c4e36060f7348/pkg/backup/delete_helpers.go

Including this package directly evolves vendoring core velero methods, so assuming that is the only place where DeleteRequest will be used, I decided to include it here.

Reply:
Actually, there is no need for those, as the labels will be set during request processing: https://github.com/konveyor/mig-controller/blob/60c8bcae42e10d89938137bb58b0cf8b6ad4d811/vendor/github.com/vmware-tanzu/velero/pkg/controller/backup_deletion_controller.go#L238-L258